src/test/regress/expected/intermediate_result_pruning.out

```diff
@@ -933,6 +933,66 @@ DEBUG:  Subplan XXX_1 will be sent to localhost:xxxxx
      0
 (1 row)
 
+-- test case for issue #3556
+CREATE TABLE accounts (id text PRIMARY KEY);
+DEBUG:  building index "pg_toast_xxxxx_index" on table "pg_toast_xxxxx" serially
+DEBUG:  CREATE TABLE / PRIMARY KEY will create implicit index "accounts_pkey" for table "accounts"
+DEBUG:  building index "accounts_pkey" on table "accounts" serially
+CREATE TABLE stats (account_id text PRIMARY KEY, spent int);
+DEBUG:  building index "pg_toast_xxxxx_index" on table "pg_toast_xxxxx" serially
+DEBUG:  CREATE TABLE / PRIMARY KEY will create implicit index "stats_pkey" for table "stats"
+DEBUG:  building index "stats_pkey" on table "stats" serially
+SELECT create_distributed_table('accounts', 'id');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT create_distributed_table('stats', 'account_id', colocate_with => 'accounts');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+INSERT INTO accounts (id) VALUES ('foo');
+INSERT INTO stats (account_id, spent) VALUES ('foo', 100);
+SELECT *
+FROM
```

Comment:
When I undo the code changes, this test still passes. So, it is probably not covering the exact case mentioned in the issue due to different number workers/shards etc.

Can you make sure that the test is affected by the code change?

Reply:
When I revert the code change, the tests fail on my local setup. I could not reproduce this problem.

I tested it 4 times in total. Twice on PG12.2, and twice on PG11.7

Reply:
To close the loop, here is the issue:

If you use `check-base` target, there are secondaries and unavailable worker nodes in the cluster. However when running `check-multi` all the nodes are primaries.

The bug can not be reproduced when there are inactive/secondary/etc nodes.